Api interfaces modifications

zakaBouj opened the pull request.

Gord310 commented:
Ok first, I think that this would be a good Idea ... but when we are doing the UI we should add a input field for the Key.
Second ... pleas don't just merge the main branch into my working branch. I had a small heart attack when I couldn't push becaus of differences XD

Gord310 reviewed (approved):
LGTM. You can merge this if you want @zakaBouj 
Just know that I moved my test to a dedicated Api Interface test file

zakaBouj commented:
Hey @Gord310 ,

I turned your `DroneApiInterfaceTest.java` into a JUnit 5 unit test and have added a test root directory that is known to Maven, so we can comfortably run all tests with the IntelliJ run button (top right) as you can see in the picture I added. I also changed the way the token is accessed; now it gets it from the environment variable, so you have to add that. I added a short description here: [https://github.com/Code-Hauptwache/DroneDeck?tab=readme-ov-file#environment-setup](https://github.com/Code-Hauptwache/DroneDeck?tab=readme-ov-file#environment-setup).
![image](https://github.com/user-attachments/assets/7f0e9193-f5a2-45a4-82a1-420f3b46e048)
